### PR TITLE
community: fix neo4j from_existing_graph

### LINKED ISSUE #24401: Neo4jVector doesn't work well with HuggingFaceEmbeddings when reusing the graph
### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the LangChain documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I am sure that this is a bug in LangChain rather than my code.
- [X] The bug is not resolved by updating to the latest stable version of LangChain (or the specific integration package).

### Example Code

```python
from langchain_community.vectorstores import Neo4jVector
from langchain_huggingface import HuggingFaceEmbeddings

embeddings = HuggingFaceEmbeddings(
    model_name="sentence-transformers/all-mpnet-base-v2"
)
self.existing_graph_parts = Neo4jVector.from_existing_graph(
    embedding=embeddings,
    url=uri,
    username=username,
    password=password,
    node_label="part",
    text_node_properties=["name"],
    embedding_node_property="embedding",
)
```

### Error Message and Stack Trace (if applicable)

```
Traceback (most recent call last):
  File "D:\graph_rag.py", line 133, in <module>
    graph_rag = GraphRag()
                ^^^^^^^^^^
  File "D:\graph_rag.py", line 44, in __init__
    self.existing_graph_parts = Neo4jVector.from_existing_graph(
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\syh\AppData\Local\Programs\Python\Python312\Lib\site-packages\langchain_community\vectorstores\neo4j_vector.py", line 1431, in from_existing_graph
    text_embeddings = embedding.embed_documents([el["text"] for el in data])
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\syh\AppData\Local\Programs\Python\Python312\Lib\site-packages\langchain_huggingface\embeddings\huggingface.py", line 87, in embed_documents
    embeddings = self.client.encode(
                 ^^^^^^^^^^^^^^^^^^^
  File "C:\Users\syh\AppData\Local\Programs\Python\Python312\Lib\site-packages\sentence_transformers\SentenceTransformer.py", line 565, in encode
    if all_embeddings[0].dtype == torch.bfloat16:
       ~~~~~~~~~~~~~~^^^
IndexError: list index out of range
```

### Description

Sorry for my poor English!

When I run the code first time, it worked well.

But when I rerun the code, it run error as above.

I think it error because all nodes has its embedding already, so when run the code in the lib below:
file: langchain_community\vectorstores\neo4j_vector.py
from line 1421

```python
        while True:
            fetch_query = (
                f"MATCH (n:`{node_label}`) "
                f"WHERE n.{embedding_node_property} IS null "
                "AND any(k in $props WHERE n[k] IS NOT null) "
                f"RETURN elementId(n) AS id, reduce(str='',"
                "k IN $props | str + '\\n' + k + ':' + coalesce(n[k], '')) AS text "
                "LIMIT 1000"
            )
            data = store.query(fetch_query, params={"props": text_node_properties})
            text_embeddings = embedding.embed_documents([el["text"] for el in data])
```

This code will fetch some nodes which don't have embedding_node_property. Since all nodes in my neo4j already have embedding, so the data is a empty list.
Then in the code following, 0 is out of an empty list's index.
file: sentence_transformers\SentenceTransformer.py
from line 563

```python
        elif convert_to_numpy:
            if not isinstance(all_embeddings, np.ndarray):
                if all_embeddings[0].dtype == torch.bfloat16:
                    all_embeddings = np.asarray([emb.float().numpy() for emb in all_embeddings])
                else:
                    all_embeddings = np.asarray([emb.numpy() for emb in all_embeddings])
```

That's where the error happened.

I have got the answer from the bot, but I still think it is bug which needs to be fixed!

Thanks!

### System Info

langchain==0.2.6
langchain-community==0.2.6
langchain-core==0.2.10
langchain-huggingface==0.0.3
langchain-openai==0.1.10
langchain-text-splitters==0.2.2

windows 11
python3.12

### PR DESCRIPTION
**Pull request by jigsawlabs-student**
Fixes Neo4JVector.from_existing_graph integration with huggingface

Previously threw an error with existing databases, because from_existing_graph query returns empty list of new nodes, which are then passed to embedding function, and huggingface errors with empty list.

Fixes [24401](https://github.com/langchain-ai/langchain/issues/24401)

### CODE REVIEW

**Comment by jigsawlabs-student** [comment]
Also @tomasonjo Noticed that some of the integration tests with calls to`similarity_search` are flaky, as the embeddings are mocked out, but then look for an exact match on the text.

See [example](https://github.com/langchain-ai/langchain/blob/master/libs/community/tests/integration_tests/vectorstores/test_neo4jvector.py#L140)

Can put in fix where just looks to see that it returns type Document, and does not look for text match, as the tests are not testing the similarity search functionality, but rather other cases.

**Comment by JeffKatzy** [comment]
@eyurtsev fixes a Neo4j/huggingface integration issue.  Let me know if it looks good -- thanks!

**Comment by tomasonjo** [comment]
This doesnt really fix anything

**Comment by JeffKatzy** [comment]
@tomasonjo Can you provide more info?  Do you mean it doesn't fix [this issue](https://github.com/langchain-ai/langchain/issues/24401) or that fixing it is not needed.

File: libs/community/langchain_community/vectorstores/neo4j_vector.py [hunk 1]
```diff
@@ -1449,6 +1449,8 @@ def from_existing_graph(
                 "LIMIT 1000"
             )
             data = store.query(fetch_query, params={"props": text_node_properties})
+            if not data:
```
Comment (tomasonjo) [inline libs/community/langchain_community/vectorstores/neo4j_vector.py:1452]: This doesn't really solve anything related to huggingface

Comment (JeffKatzy) [reply libs/community/langchain_community/vectorstores/neo4j_vector.py:1452]: Ok, so I just re-ran the tests in [here](https://github.com/langchain-ai/langchain/pull/24912/files#diff-97f27032565f976999bcb3fdf1a3f44a34ac8ae47f6cb3c7ad9fa5320ecd022d) and replaced the 

`FakeHuggingFaceEmbeddings` with 
```
embeddings = HuggingFaceEmbeddings(
        model_name="sentence-transformers/all-mpnet-base-v2"
    )
```
and commented out the `if not data:` line.

I still get the following error:

```bash
langchain_community/vectorstores/neo4j_vector.py:1454: in from_existing_graph
    text_embeddings = embedding.embed_documents([el["text"] for el in data])
/home/vscode/langchain-py-env/lib/python3.11/site-packages/langchain_huggingface/embeddings/huggingface.py:87: in embed_documents
    embeddings = self.client.encode(
>               if all_embeddings[0].dtype == torch.bfloat16:
E               IndexError: list index out of range

/home/vscode/langchain-py-env/lib/python3.11/site-packages/sentence_transformers/SentenceTransformer.py:565: IndexError
```

So the issue people are running into is that when [this query](https://github.com/langchain-ai/langchain/blob/master/libs/community/langchain_community/vectorstores/neo4j_vector.py#L1451) returns an empty list, which it does when all documents have already been embedded, then HF throws an error as `embed_documents` looks for `all_embeddings[0]`, resulting in the `IndexError`.  

So not proceeding with embeddings when it's embedding already occurred results in avoiding the `IndexError` without apparent side effects.  Both the error and fix appear to align with what others have observed [here](https://github.com/langchain-ai/langchain/issues/24401).

Comment (tomasonjo) [reply libs/community/langchain_community/vectorstores/neo4j_vector.py:1452]: Then add just this line and nothing else.

Comment (JeffKatzy) [reply libs/community/langchain_community/vectorstores/neo4j_vector.py:1452]: Ok, so remove the tests?

Comment (tomasonjo) [reply libs/community/langchain_community/vectorstores/neo4j_vector.py:1452]: sure

Comment (JeffKatzy) [reply libs/community/langchain_community/vectorstores/neo4j_vector.py:1452]: Ok, let me give the tests another shot by using the real huggingface embeddings, as I would like to show the issue.  But if it doesn't look good, I'll just remove the tests.

Comment (JeffKatzy) [reply libs/community/langchain_community/vectorstores/neo4j_vector.py:1452]: Ok, I wasn't sure how to best update the `FakeHuggingFaceEmbeddings` or if it was ok to reference use the real embeddings, so just removed the test.

Comment (tomasonjo) [reply libs/community/langchain_community/vectorstores/neo4j_vector.py:1452]: ok, now it's GTG... Thanks

**Review by tomasonjo — APPROVED** [review]
Thanks

**Review by efriis — APPROVED** [review]

**Pull request merged**